apps/angular/5-crud-application/src/components/to-do/to-do.component.ts

```diff
@@ -0,0 +1,53 @@
+import { CommonModule } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+  SkipSelf,
+} from '@angular/core';
+import { MatProgressSpinner } from '@angular/material/progress-spinner';
+import { randText } from '@ngneat/falso';
+import { TodoDetails } from './../../model/user';
+import { CrudSpecialService } from './../../services/crud-special.service';
+
+@Component({
+  selector: 'app-to-do',
+  standalone: true,
+  imports: [CommonModule, MatProgressSpinner],
+  templateUrl: `./to-do.component.html`,
+  styles: ``,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ToDoComponent implements OnInit {
+  @Input() todo!: TodoDetails;
+  @Output() todoChange = new EventEmitter<TodoDetails>();
+  @Output() todoDelete = new EventEmitter<TodoDetails>();
+  //@Output() update = new EventEmitter<TodoDetails>();
+  public loading = true;
+
+  constructor(@SkipSelf() private todoService: CrudSpecialService) {}
+
+  ngOnInit(): void {
+    this.loading = this.todo == undefined;
+  }
+
+  deleteTodo(): void {
+    this.loading = true;
+    this.todoService.deleteTodo(this.todo).subscribe(() => {
+      this.todoDelete.emit(this.todo);
+    });
+  }
+
+  updateTodo(): void {
+    this.loading = true;
+    this.todo.title = randText();
+    this.todoService.updateTodo(this.todo).subscribe((todoUpdated) => {
+      this.todo.title = todoUpdated.title;
+      this.loading = false;
+      this.todoChange.emit(this.todo);
+    });
```

Comment:
you code will become very difficult to maintain.

You should have one source of trust, and it should be inside your service. 
You have your todo list inside the service as a signal, you update, delete, set and your component listen to that signal and will update reactively the view. 

Your app is working magically because your parent component is not onPush. When you update your property using imperative coding like `this.todp.title = xxx`, the app will not trigger a new change detection cycle, but because you emit your value to the parent and the parent recreates a signal, the app is refreshing but that's not very maintainable.